dot-net-sdk/dto/Allocation.cs

```diff
@@ -1,5 +1,26 @@
+using System.ComponentModel;
+using Newtonsoft.Json;
+
 namespace eppo_sdk.dto;
 
-public record Allocation(string key, List<Rule> rules, List<Split> splits, bool doLog, DateTime? startAt, DateTime? endAt)
-{
+public class Allocation 
+{   
+    public string Key {get;}
+    public List<Rule> Rules {get;} = new List<Rule>();
+    public List<Split> Splits {get;}
+    public DateTime? StartAt {get;} = null;
+    public DateTime? EndAt {get;} = null;
+
+    [DefaultValue(true)]
+    [JsonProperty(DefaultValueHandling = DefaultValueHandling.Populate)]
```

Comment:
Missed setting default value for `doLog` to true with the UFC implementation. Added a test to verify assignment logging in `EppoClientTest`. Added a test case in `BanditClientTest` which verifies that `DoLog` is set to `true` when it is missing in the underlying json payload.

Reply:
I only hit this when testing for assignment logging in the bandit client test using the [test data file](https://github.com/Eppo-exp/sdk-test-data/blob/main/ufc/bandit-flags-v1.json)

Reply:
right yeah really we should just update that file